### PR TITLE
Fix/field label as html

### PR DESCRIPTION
**Pull request by gerjanvangeest**

### CODE REVIEW

**Comment by tlouisse** [comment]
LGTM 👍  Should we actually do the same for helpText?

File: packages/field/src/FormControlMixin.js [hunk 1]
```diff
@@ -53,6 +53,34 @@ export const FormControlMixin = dedupeMixin(
...
+        return (
+          (this.querySelector('[slot="label"]') &&
+            this.querySelector('[slot="label"]').textContent) ||
+          this.__label
+        );
+      }
+
+      set label(newValue) {
+        const oldValue = this.label;
+        this.__label = newValue;
```
Comment (jorenbroekema) [inline packages/field/src/FormControlMixin.js]: we have protected _labelNode but private __label? Shouldn't they have the same access?

**Comment by jorenbroekema** [comment]
What I am missing is an initially failing test that now succeeds, where we test that label with specifically html inside of it, doesn't work, and that now it does work. It's a bit unclear to me how this PR does anything to the "html inside label" feature.

File: packages/field/src/FormControlMixin.js (hunk 1 above)
Comment (gerjanvangeest) [reply packages/field/src/FormControlMixin.js]: `label` is the public api, `__label` is just needed for internal use.

**Pull request merged**